src/main/java/com/_up/megastore/services/implementations/ProductService.java

```diff
@@ -56,5 +57,12 @@ private String saveProductImage(MultipartFile multipartFile) {
     private Product getVariantOf(UUID variantOfId) {
         return variantOfId != null ? findProductByIdOrThrowException(variantOfId) : null;
     }
+    public ProductResponse updateProduct(UpdateProductRequest updateProductRequest, MultipartFile multipartFile){
+        Category category = categoryService.findCategoryByIdOrThrowException(UpdateProductRequest.categoryId());
```

Comment:
Acá estás queriendo usar el categoryId que te llega de la request, pero el orden en el que se definen las cosas en el método es

<TipoDeAcceso> <TipoDeRetorno> <nombreMetodo> (TipoDeVariable nombreVariable) {}

por lo tanto, deberias usar updateProductRequest en vez de UpdateProductRequest

```suggestion
        Category category = categoryService.findCategoryByIdOrThrowException(updateProductRequest.categoryId());
```